src/derive/curve.rs

```diff
@@ -232,15 +279,20 @@ macro_rules! new_curve_impl {
 
                         fn from_uncompressed_unchecked(bytes: &Self::Uncompressed) -> CtOption<Self> {
                             let bytes = &bytes.0;
-                            let infinity_flag_set = Choice::from((bytes[[< $name _UNCOMPRESSED_SIZE >] - 1] >> 6) & 1);
-                            // Attempt to obtain the x-coordinate
+                            let infinity_flag_set = if $spare_bits == 1 {
+                                // With 1 spare bit there is no infinity flag, so we just rely
+                                // on the x, y coordinates to be set to 0.
+                                Choice::from(1u8)
+                            } else {
+                                Choice::from((( bytes[[<$name _FLAT_BYTE_INDEX >]] & IS_IDENTITY_MASK) >> IS_IDENTITY_SHIFT) )
+                            };
+
```

Comment:
Question @davidnevadoc 
I read that the uncompressed form of 0 spare bits do NOT have extra bytes for flags.
But, this code computes the `infinity_flag_set` from the 0 spare bits point.
It means that the MSB of `y` point is used for computation.
I think it can cause the error. 
Is it right?

Reply:
Hmm, there is also a mismatch between the proposal and impl, for 2 spare bits case.
In case of the 2 spare bits, the uncompressed form of `P` stores the `ident` flag in MSB of `P.x`.
But, the impl is different.
It computes the `identity_flag_set` from the MSB of `P.y`.
Since the `is_ident` is computed as `identity_flag_set & x.is_zero() & y.is_zero()`, I believe it can cause the error. https://github.com/privacy-scaling-explorations/halo2curves/pull/141/files#diff-a8cb01ec1b5a3d6c72d164a54bdf6a06049e53be96cb2d9be434618d5a8fec15R306